doc/holochain_101/src/zome/define_zome.md

```diff
@@ -28,33 +28,33 @@ What are Rust macros? Generally speaking, they are code that will actually gener
 
 In the case of Zome development, it was discovered that much code could be saved from being written, by encapsulating it in a macro.
 
-That is how `define_zome!` came about. It is a Rust macro imported from the HDK which must be used for every Zome (unless you read the source code for it yourself and write something that behaves the same way!)
+That is how `#[zome]` came about. It is a Rust macro imported from the HDK which must be used for every Zome (unless you read the source code for it yourself and write something that behaves the same way!)
 
 The following is technically the most minimalistic Zome that could be implemented. It does nothing, but still conforms to the expectations Holochain has for a Zome.
 
 ```rust
 #[macro_use]
 extern crate hdk;
 
-define_zome! {
-    entries: []
-
-    init: || {
+#[zome]
+mod my_zome {
+    
+    #[init]
+    fn init() -> ZomeApiResult<()> {
         Ok(())
     }
 
-    validate_agent: |validation_data : EntryValidationData::<AgentId>| {
+    #[validate_agent]
+    fn validate_agent(validation_data : EntryValidationData::<AgentId>) -> ZomeApiResult {
         Ok(())
```

Comment:
Important but not urgent:
Showing the simplest useful example of how `validate_agent()` would be used (e.g. blacklist of dis-allowed agents or something?) would be better than just returning `Ok(())` for all calls in an introductory example like this.

Reply:
I have a pretty good example of that in [init.md](https://github.com/holochain/holochain-rust/blob/06bbe308143325282583dbdeb2ad148208853201/doc/holochain_101/src/zome/init.md). Do you think this is ok or should I move it to here?

Reply:
I'd recommend either
- showing simple example code that helps the reader understand how to use `#[validate_agent]` effectively
- ensure `#[validate_agent]` is optional with a reasonable default and discuss its use as a more advanced topic (the example in `init.md` is great)

I'd recommend the same for `#[init]`.